rule/rule.go

```diff
@@ -46,6 +46,8 @@ type Remediation struct {
 	// Text in markdown format for humans.
 	Markdown string `json:"markdown"`
 	// Effort to remediate.
+	Branch string            `json:"branch"`
+	Repo   string            `json:"repo"`
```

Comment:
They are needed for the remediation output. Ideally we should use `remediation.RemediationMetadata`, but that currently gives circular imports (AFAIR because remediation imports "/checker").

Reply:
If the fields aren't need any more due to https://github.com/ossf/scorecard/pull/3816#discussion_r1516796096, I prefer they be removed

Reply:
https://github.com/ossf/scorecard/pull/3816/commits/713839919156512f232788a165766d058fcceafd